drivers/renesas/rcar/board/board.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2015-2018, Renesas Electronics Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <stdint.h>
+#include <iic_dvfs.h>
+#include "board.h"
+
+#ifndef BOARD_DEFAULT
+#if (RCAR_LSI == RCAR_E3)
+#define BOARD_DEFAULT		(BOARD_EBISU << BOARD_CODE_SHIFT)
+#else
+#define BOARD_DEFAULT		(BOARD_SALVATOR_X << BOARD_CODE_SHIFT)
+#endif
+#endif
+
+#define ARRAY_SIZE(__x) 	(sizeof(__x)/sizeof(__x[0]))
+
+#define SLAVE_ADDR_EEPROM	(0x50)
+#define	REG_ADDR_BOARD_ID	(0x70)
+#define BOARD_CODE_MASK		(0xF8)
+#define BOARD_REV_MASK		(0x07)
+#define BOARD_CODE_SHIFT	(0x03)
+#define BOARD_ID_UNKNOWN	(0xFF)
+
+const char *g_board_tbl[] = {
+	[BOARD_SALVATOR_X] = "Salvator-X",
+	[BOARD_KRIEK] = "Kriek",
+	[BOARD_STARTER_KIT] = "Starter Kit",
+	[BOARD_SALVATOR_XS] = "Salvator-XS",
+	[BOARD_EBISU] = "Ebisu",
+	[BOARD_STARTER_KIT_PRE] = "Starter Kit",
+	[BOARD_EBISU_4D] = "Ebisu-4D",
+	[BOARD_UNKNOWN] = "unknown"
+};
+
+int32_t rcar_get_board_type(uint32_t * type, uint32_t * rev)
+{
+	int32_t ret = 0;
+	const uint8_t board_tbl[][8] = {
+		[BOARD_SALVATOR_X] = {0x10U, 0x11U, 0xFFU, 0xFFU,
+				      0xFFU, 0xFFU, 0xFFU, 0xFFU},
+		[BOARD_KRIEK] = {0x10U, 0xFFU, 0xFFU, 0xFFU,
+				 0xFFU, 0xFFU, 0xFFU, 0xFFU},
+		[BOARD_STARTER_KIT] = {0x10U, 0xFFU, 0xFFU, 0xFFU,
+				       0xFFU, 0xFFU, 0xFFU, 0xFFU},
+		[BOARD_SALVATOR_XS] = {0x10U, 0xFFU, 0xFFU, 0xFFU,
+				       0xFFU, 0xFFU, 0xFFU, 0xFFU},
+		[BOARD_EBISU] = {0x10U, 0xFFU, 0xFFU, 0xFFU,
+				 0xFFU, 0xFFU, 0xFFU, 0xFFU},
+		[BOARD_STARTER_KIT_PRE] = {0x10U, 0x10U, 0xFFU, 0xFFU,
```

Comment:
Are the **two** first entries in the _KIT_PRE meant to be 0x10U? Just checking in case it is a typo and the second was meant to be 0x11 or 0x20.